src/cryptography/x509/ocsp.py

```diff
@@ -163,7 +163,73 @@ def extensions(self) -> x509.Extensions:
         """
 
 
+class OCSPResponseIterator(metaclass=abc.ABCMeta):
```

Comment:
This type isn't actually an iterator, this seems to be a single response and then `response_iter` returns a `typing.Iterator[ThisType]`?

Reply:
It's an iterator, in that it implements `__next__` and `__iter__`.  The root of all my pain is that my understanding of PyO3 is that I have to return a pyclass object, which can't contain any lifetimes.  This means I can't just return `SingleResponse<'_>`, or turn it into a pyclass.  To solve this problem, I have the iterator object with the index on it.  Each time you call `__next__`, it increments the index, then when you call the property methods (e.g. `serial_number`), it burrows into contents, finds the appropriate `SingleResponse`, extracts the requested field in a way Python can handle, and returns it.

If I understand what you're suggesting, you want __next__ do something like:

```python
let next_resp = slf
            .contents
            .borrow_basic_response()
            .as_ref()
            .unwrap()
            .tbs_response_data
            .responses
            .unwrap_read()
            .next()
            .unwrap();
```

Then use `next_resp` to create a new pyclass object.  I can get close to this, since the `__next__` method has `slf.py()` to get the `pyo3::Python` object, but when I try to advance`...tbs_response_data.responses` it gives me an error about being unable to borrow the result of `unwrap_read()` as mutable.

If you have any advice on how to get `tbs_response_data.responses` into a state where I can `next()` through it, that would be great.

Reply:
The ABC doesn't contain `__next__`, that's what's confusing me. This seems to be an ABC for `SingleResponse`.